src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts

```diff
@@ -160,10 +162,17 @@ export class InlineChatController implements IEditorContribution {
 		this._ctxSupportIssueReporting = CTX_INLINE_CHAT_SUPPORT_ISSUE_REPORTING.bindTo(contextKeyService);
 		this._zone = new Lazy(() => this._store.add(_instaService.createInstance(InlineChatZoneWidget, this._editor)));
 
+		this._store.add(this._inlineChatSessionService.onDidChangeSessionEnablementStatus(() => {
```

Comment:
In here I would get all providers (and listen for changes when they come and go) and for each provider I would listen if they change their enablement. Then I would pick the first provider that is enabled and create a session with that.

Reply:
There is also the interesting (corner) case of a provider disabling itself while in-use. In that case we can terminate the session similar to the logic on line 186

Reply:
We call `createSession` which delegates to `IInlineChatSessionService` and that is what picks the first provider to create the session for. Are you saying this logic should move to the controller and the controller should pass in which provider it wants created?